### ISSUE TITLE
Exception: dependency on non-existent parameter partkeepr.parts.internalpartnumberunique

### ISSUE THREAD

**Issue by olewolf** [partkeepr/PartKeepr#725]
Commit 098acfe00d09246648ad61bca7b9dd389ccb3de3 (the most recent one as of this writing) appears to have broken my install. The web server log shows the following exception:

[Sun Aug 21 07:43:56.792971 2016] [:error] [pid 15812] [client 2001:xxxxxxxxxxxx] PHP Fatal error:  Uncaught Symfony\\Component\\DependencyInjection\\Exception\\ParameterNotFoundException: The service "partkeepr.part_service" has a dependency on a non-existent parameter "partkeepr.parts.internalpartnumberunique". in /usr/local/share/PartKeepr/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php:94\nStack trace:\n#0 /usr/local/share/PartKeepr/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php(214): Symfony\\Component\\DependencyInjection\\ParameterBag\\ParameterBag->get('partkeepr.parts...')\n#1 /usr/local/share/PartKeepr/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php(185): Symfony\\Component\\DependencyInjection\\ParameterBag\\ParameterBag->resolveString('%partkeepr.part...', Array)\n#2 /usr/local/share/PartKeepr/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php(175): Symfony\\Component\\DependencyInjection\\ParameterBag\\ParameterBag->resolveValue('% in /usr/local/share/PartKeepr/vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/ParameterBag/ParameterBag.php on line 94

**Comment by felicitus** [comment]
Always copy newly added parameters as per https://wiki.partkeepr.org/wiki/Running_PartKeepr_from_GIT

**Comment by olewolf** [comment]
I actually followed this upgrade procedure but the error remains. Please reopen issue.

**Comment by felicitus** [comment]
I just found and fixed the issue - please upgrade git. Please put all required information (like you followed the procedure) in the issue. I assumed that you didn't, because that's the most common problem when people upgrading using GIT ;)

**Comment by ole-modos** [comment]
Sounds great--and sorry I didn't remember to add that I had followed the upgrade procedure.

I wanted to checkout the development branch again, and I feel a little stupid for asking this, but what was the current development branch named? (If it's "HEAD," then the fix is working but I'm not sure if this is the correct branch.)

**Comment by felicitus** [comment]
It's git checkout master ;)

**Comment by olewolf** [comment]
Hm, in that case the error seems to persist. I'm getting exactly the same error message as before. I'm receiving no error reports by composer during the upgrade.

I'm not sure if this means anything, but the only recent change seem to be a modification to app/check.php and app/SymfonyRequirements.php, and the git log only says that ChangeLog has been updated.

**Comment by felicitus** [comment]
have you done a git pull while in the master branch?

**Comment by felicitus** [comment]
Something is very strange indeed with git - let me check, I'll give you a notification once I find out what the issue is

**Comment by olewolf** [comment]
Yes; I'm on the most recent pull from the master branch:

$ git branch
* master
$ git pull
Already up-to-date.

It looks like check.php and SymfonyRequirements.php are actually being updated when I run the "composer install" section of the upgrade procedure so I'm not sure what's being updated. Looking at the github commit log, only the changelog has been committed and push since three days ago. Since reported the error just two days ago, could it be you accidentally didn't merge the fix into master or something?

**Comment by felicitus** [comment]
Well it seems that I only pushed the tags for the new version without actually pushing the commits themselves ;) I had the exact same issue as you when I just tried to update on my personal system. After I pushed the commits, it works now. Sorry for the problems

**Comment by olewolf** [comment]
Git glitches like that do tend to happen occasionally. :)

Alas, in spite of your efforts, the error is still there on my end (after git pulling and updating, of course).

**Comment by felicitus** [comment]
Can you give me the output of `cat src/PartKeepr/SetupBundle/Services/ConfigSetupService.php |grep -i internalpart`?

**Comment by olewolf** [comment]
Of course:

```
$ cat src/PartKeepr/SetupBundle/Services/ConfigSetupService.php |grep -i internalpart
            'partkeepr.parts.internalpartnumberunique' => false,
```

**Comment by olewolf** [comment]
That's a little hard to tell but it seems I never even get to the login prompt. Apache just bombs with a 500 error and the error message (and no other messages) appears in the error log.

**Comment by felicitus** [comment]
So it works during setup but fails after opening the web frontend?

**Comment by olewolf** [comment]
My bad, sorry: I just managed to think, "gee, this looks like a database schema that hasn't been updated..." when you mentioned running the web setup. I had only run through copying the settings files and the composer updates this time and hadn't run the setup page. (I suppose that's what I get from writing a bash script for the upgrades, so now I've added a message for myself in the script that reminds me to run setup.)

In any event - it works now. Thanks for your prompt fix and your help!
